provider/cmd/pulumi-resource-aws-apigateway/apigateway/api.ts

```diff
@@ -821,14 +833,104 @@ function createSwaggerSpec(
         name,
         swagger,
         route,
-        apiAuthorizers
       );
     } else {
       addRawDataRouteToSwaggerSpec(parent, name, swagger, route);
     }
   }
 }
 
+/**
+ * Checks for duplicate authorizers in the given array.
+ * @param authorizers - The array of authorizers to check.
+ * @returns A pulumi.Output containing the name of the duplicate authorizer, if found; otherwise, undefined.
+ */
+function checkDuplicateAuthorizer(
+  authorizers: Authorizer[]
+): pulumi.Output<string | undefined> {
+  const namedAuthorizers: Record<string, Authorizer> = {};
+  const duplicates = authorizers.map((auth) => {
+    if (!auth.authorizerName) {
+      return undefined;
+    }
+
+    if (namedAuthorizers[auth.authorizerName]) {
+      return authorizerEquals(namedAuthorizers[auth.authorizerName], auth).apply((equal) => {
+        if (!equal) {
```

Comment:
How does this work? I would expect this to need to be 

```
if (equal) {
  return auth.authorizerName;
}
```

Because we want to return the name of the duplicate authorizer when they are equal right?

Reply:
Maybe my naming here was a bit misleading. We're checking for authorizers that have the same name, but different configuration. I'm gonna add some more description to the docstring

Reply:
@corymhall I updated it. Let me know what you think

Reply:
ahh gotcha. I'm slow this morning 😀